Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -1377,6 +1378,25 @@ void WebProcessProxy::renderTreeAsText(WebCore::ProcessIdentifier destinationPro
 
 bool WebProcessProxy::canBeAddedToWebProcessCache() const
 {
+    // Don't cache WebContent processes that are running. Before being considered for caching,
+    // WebContent processes will be suspended and inserted into the Back/Forward cache.
+    // If the process has not suspended by now, there is likely an assertion leak, that keeps
+    // it running. It is better to terminate the process, than keep it running by adding it to
+    // the cache.
+    if (throttler().currentState() != ProcessThrottleState::Suspended) {
```

Comment:
As mentioned in my earlier review, I disagree with this. Why aren't we caching processes that are not suspended? This seems racy. Processes enter the process cache when they are no longer used but getting suspended is something that happens asynchronously and may take an arbitrary amount of time.

Reply:
Ah, yes, good catch, this is racy. I will attempt to address this in the next patch.